node/silkworm/downloader/internals/db_tx.hpp

```diff
@@ -202,6 +203,45 @@ class Db::ReadOnlyAccess::Tx {
     }
 
     BlockNum read_stage_progress(const char* stage_name) { return db::stages::read_stage_progress(txn, stage_name); }
+
+    // see Erigon's HeadersUnwind method for the implementation
+    std::tuple<BlockNum, Hash> header_with_biggest_td(const std::set<Hash>* bad_headers = nullptr) {
+        BlockNum max_block_num = 0;
+        Hash max_hash;
+        BigInt max_td = 0;
+
+        auto td_cursor = db::open_cursor(txn, db::table::kDifficulty);
+
+        auto find_max = [bad_headers, &max_block_num, &max_hash, &max_td](mdbx::cursor&,
+                                                                          mdbx::cursor::move_result& result) -> bool {
+            ByteView key = db::from_slice(result.key);
+            ByteView value = db::from_slice(result.value);
+
+            if (key.size() != sizeof(uint64_t) + kHashLength) {
```

Comment:
PR #488 introduces `SILKWORM_ASSERT`, which aborts the program on assert failure even in release mode. Here aborting the program in the case of the wrong key size is probably the only thing we can do, since it's unlikely that we can recover from corrupted database. So I'd replace exception throwing with
```
SILKWORM_ASSERT(key.size() == sizeof(BlockNum) + kHashLength);
```

Reply:
Sure! I added a commit with the proposed change. Also I will substitute logic_error with SILKWORM_ASSERT in other places, where it makes sense, in future PR